util.py

```diff
@@ -409,7 +410,7 @@ def __init__(self, duration, max_mem_mb = None):
 
         #Set the duration
         self._duration = int(duration)
-        signal.signal(signal.SIGALRM, Watchdog.self_destruct)
+        self._timer = threading.Timer(self._duration, Watchdog.self_destruct)
```

Comment:
Actually, `.reset` is called when the `Watchdog` is created: https://github.com/DataDog/dd-agent/blob/d10e5f621186085a22f51679caf92906a2c1fd14/agent.py#L172